### PR TITLE
Redux toolkit

### PR DESCRIPTION
**Pull request by mohammed-sas**
### Features
1. useContext + useReducer has been converted to Redux State Management

### CODE REVIEW

File: src/redux/slices/postSlice.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+export const postSlice = createSlice({
+  name: "posts",
+  initialState,
+  reducers: {
+    clearPosts:(state)=>{
+        state.posts=[];
+        state.loading=false;
+    }
+  },
+  extraReducers: {
```
Comment (Kevin-Solomon) [inline src/redux/slices/postSlice.js:27]: why not use the builder callback notion in extraReducer.

**Review by Kevin-Solomon — APPROVED** [review]
The code looks clean.
Amazing job with the refactoring.

**Review by Kevin-Solomon — CHANGES_REQUESTED** [review]
The like handler is throwing an error.
image

**Review by Kevin-Solomon — CHANGES_REQUESTED** [review]
Also, when a user adds a post the number of posts, is not updated on the profile page.
image

**Review by Kevin-Solomon — CHANGES_REQUESTED** [review]
onClick of the ellipsis the app throws an Error

**Review by Kevin-Solomon — APPROVED** [review]

**Review by BISINIGISHIVANI — APPROVED** [review]

**Pull request merged**